File: Data Structures & Algorithms/concatenation-of-array/submission-0.cpp
class Solution {
public:
    vector<int> getConcatenation(vector<int>& nums) {
        std::vector<int> ans;
        for (int i = 0; i < 2; ++i) {
            for (auto e : nums) {
                ans.push_back(e);
            }
        }
        return ans;
    }
};

File: Data Structures & Algorithms/subsets-ii/submission-0.py
class Solution:
    def subsetsWithDup(self, nums: list[int]) -> list[list[int]]:
        subsets: list[list[int]] = [[]]
        for n in sorted(nums):
            variants = subsets.copy()
            subsets = []
            for variant in variants:
                left, right = self.recur(variant, n)
                subsets.append(left)
                subsets.append(right)
        return [list(subset) for subset in set(map(tuple, subsets))]

    def recur(self, prev: list[int], next: int) -> tuple[list[int], list[int]]:
        right = prev.copy()
        right.append(next)
        return prev.copy(), right